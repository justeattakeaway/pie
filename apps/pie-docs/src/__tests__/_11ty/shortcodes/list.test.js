const list = require('../../../_11ty/shortcodes/list');
const listTypes = require('../../../_data/listTypes');

describe('list.js', () => {
    const items = ['a', 'b', 'c', 'd'];

    it.each([
        'ordered',
        'pill'
    ])('should return the expected HTML', type => {
        // act
        const result = list({ type, items });

        // assert
        expect(result).toMatchSnapshot();
    });

    it('should return the expected HTML for an icon list', () => {
        // act
        const result = list({
            type: 'icon', iconName: 'close-circle-filled', iconFill: 'support-positive', items
        });

        // assert
        expect(result).toMatchSnapshot();
    });

    it('should throw an error if `type` is invalid', () => {
        // arrange
        const invalidType = 'not a real type';
        const invalidTypeError = new Error(`List 'type = ${invalidType}' not recognised. Try ${Object.values(listTypes).join(', ')}`);

        // assert
        expect(() => {
            list({ type: invalidType, items });
        }).toThrow(invalidTypeError);
    });
});
