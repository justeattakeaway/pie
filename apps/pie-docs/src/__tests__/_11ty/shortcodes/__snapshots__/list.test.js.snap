// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list.js should return the expected HTML 1`] = `
"<ol class=\\"c-list c-list--ordered\\"  >
        <li class=\\"c-list-item\\">
        
        <span>a</span>
        </li><li class=\\"c-list-item\\">
        
        <span>b</span>
        </li><li class=\\"c-list-item\\">
        
        <span>c</span>
        </li><li class=\\"c-list-item\\">
        
        <span>d</span>
        </li>
    </ol>"
`;

exports[`list.js should return the expected HTML 2`] = `
"<ul class=\\"c-list c-list--pill\\"  >
        <li class=\\"c-list-item\\">
        
        <span>a</span>
        </li><li class=\\"c-list-item\\">
        
        <span>b</span>
        </li><li class=\\"c-list-item\\">
        
        <span>c</span>
        </li><li class=\\"c-list-item\\">
        
        <span>d</span>
        </li>
    </ul>"
`;

exports[`list.js should return the expected HTML for an icon list 1`] = `
"<ul class=\\"c-list c-list--icon\\"  style=\\"--icon-fill: #017a39\\";>
        <li class=\\"c-list-item\\">
        <svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" height=\\"20\\" width=\\"20\\" fill=\\"#017a39\\" class=\\"c-pieIcon c-pieIcon--closeCircleFilled u-iconFilled\\" aria-hidden=\\"true\\"><path d=\\"M8 1.219A6.781 6.781 0 1014.781 8 6.79 6.79 0 008 1.219zm2.87 8.75l-.927.927L8 8.927 6.058 10.87l-.928-.927L7.072 8 5.13 6.058l.928-.928L8 7.072 9.943 5.13l.927.928L8.928 8l1.942 1.969z\\" fill=\\"#242E30\\"></path></svg>
        <span>a</span>
        </li><li class=\\"c-list-item\\">
        <svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" height=\\"20\\" width=\\"20\\" fill=\\"#017a39\\" class=\\"c-pieIcon c-pieIcon--closeCircleFilled u-iconFilled\\" aria-hidden=\\"true\\"><path d=\\"M8 1.219A6.781 6.781 0 1014.781 8 6.79 6.79 0 008 1.219zm2.87 8.75l-.927.927L8 8.927 6.058 10.87l-.928-.927L7.072 8 5.13 6.058l.928-.928L8 7.072 9.943 5.13l.927.928L8.928 8l1.942 1.969z\\" fill=\\"#242E30\\"></path></svg>
        <span>b</span>
        </li><li class=\\"c-list-item\\">
        <svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" height=\\"20\\" width=\\"20\\" fill=\\"#017a39\\" class=\\"c-pieIcon c-pieIcon--closeCircleFilled u-iconFilled\\" aria-hidden=\\"true\\"><path d=\\"M8 1.219A6.781 6.781 0 1014.781 8 6.79 6.79 0 008 1.219zm2.87 8.75l-.927.927L8 8.927 6.058 10.87l-.928-.927L7.072 8 5.13 6.058l.928-.928L8 7.072 9.943 5.13l.927.928L8.928 8l1.942 1.969z\\" fill=\\"#242E30\\"></path></svg>
        <span>c</span>
        </li><li class=\\"c-list-item\\">
        <svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 16 16\\" height=\\"20\\" width=\\"20\\" fill=\\"#017a39\\" class=\\"c-pieIcon c-pieIcon--closeCircleFilled u-iconFilled\\" aria-hidden=\\"true\\"><path d=\\"M8 1.219A6.781 6.781 0 1014.781 8 6.79 6.79 0 008 1.219zm2.87 8.75l-.927.927L8 8.927 6.058 10.87l-.928-.927L7.072 8 5.13 6.058l.928-.928L8 7.072 9.943 5.13l.927.928L8.928 8l1.942 1.969z\\" fill=\\"#242E30\\"></path></svg>
        <span>d</span>
        </li>
    </ul>"
`;
